### ISSUE TITLE
Move exceptions to steps

### ISSUE THREAD

**Issue by sagikazarmark** [portphp/portphp#21]
There are some exceptions which are related to steps:

- MappingException
- ValidationException
- probably more

They should be moved to the steps package, shouldn't they?

**Comment by ddeboer** [comment]
Well, steps, albeit now the only, is just one possible implementation of workflow. Other implementations will need the converters and exceptions too. So I prefer to keep them in the main repo for now.

**Comment by sagikazarmark** [comment]
The mentioned exceptions are only thrown by steps. Mapping exception is obviously by MappingStep, ValidationException by ValidationStep. These exceptions have nothing to do with converters, they are only used in steps.

**Comment by ddeboer** [comment]
But what if we have another workflow that also uses or validator filters and might therefore also want to throw exceptions pertaining to those?

**Comment by sagikazarmark** [comment]
I would say, validation related things should go into a separate package.

- ValidationFilter
- ValidationStep
- ValidationException

**Comment by sagikazarmark** [comment]
Currently the Symfony Validation library is required in two packages: portphp and steps. By separating all validation logic into it's own package we remove this redundancy. Not to mention: validation is not always necessary.

**Comment by ddeboer** [comment]
> By separating all validation logic into it's own package we remove this redundancy.

True, but that introduces a dependency from portphp/validation on portphp/steps (for ValidatorStep). Is that a problem?

**Comment by slashrsm** [comment]
Closing as not planned for the current maintenance line.

There was useful discussion about where MappingException / ValidationException (and validation itself) should live, but no agreed package split or PR landed. After the 2.0 modernization we are not reorganizing exception packages unless there is a concrete, low-risk proposal with a migration path.

Feel free to reopen with a focused design if this becomes actionable.
